addons/account_facturx/models/account_move.py

```diff
@@ -209,7 +205,7 @@ def _import_facturx_invoice(self, tree):
                     invoice_line_form.price_unit = amount_total_import
 
             # Refund.
-            if self_ctx.env.context['journal_type'] == 'purchase':
+            if journal.type == 'purchase':
```

Comment:
You cannot set the type as it is required
I have made a fix here https://github.com/odoo-dev/odoo/commit/01de40032221b57acb40faf8906d1b2b54a0acd8